app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/representee/summary_for_confirmation.scala.html

```diff
@@ -0,0 +1,77 @@
+@*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import play.api.i18n.Messages
+@import play.twirl.api.Html
+
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.returns.representee.routes
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.returns.triage
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.IndividualUserType.{Capacitor, PersonalRepresentative}
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.RepresenteeAnswers.CompleteRepresenteeAnswers
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.ids.NINO
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.ids.SAUTR
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.ids.CgtReference
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.RepresenteeReferenceId.RepresenteeCgtReference
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.RepresenteeReferenceId.RepresenteeSautr
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.RepresenteeReferenceId.RepresenteeNino
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.RepresenteeReferenceId.NoReferenceId
+@import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.RepresenteeAnswers.IncompleteRepresenteeAnswers
+
+@this(
+cyaSection: uk.gov.hmrc.cgtpropertydisposalsfrontend.views.html.components.cya_section,
+cyaRow: uk.gov.hmrc.cgtpropertydisposalsfrontend.views.html.components.cya_row,
+cyaChange: uk.gov.hmrc.cgtpropertydisposalsfrontend.views.html.components.cya_change
+)
+
+@(
+        answers: IncompleteRepresenteeAnswers,
+        representativeType: Either[PersonalRepresentative.type, Capacitor.type]
+)(implicit messages: Messages)
+@id = @{
+    val IncompleteRepresenteeAnswers(Some(name), Some(id), dateOfDeath, None, false) = answers
+    id match {
+        case RepresenteeNino(NINO(value)) => value
+        case RepresenteeSautr(SAUTR(value)) => value
+        case RepresenteeCgtReference(CgtReference(value)) => value
+        case NoReferenceId => ""
+        case _ => ""
+    }
+}
+
+@name = @{
```

Comment:
it's safer to pass in the name to the template

Reply:
you'd still benefit from passing in the name as a separate parameter in the template. It'd be safer to also pass in the id as a separate parameter as well. You don't necessarily have to pass in `answers: IncompleteRepresenteeAnswers` that way, and you have a safer way to extract out the information you need.

Doing ` val Some(x) = y` is never safe so we should never have it in the code

Reply:
Sorry, I don't know how I missed that but probably I thought it was the case in the journey. where any other value could not be present But shouldn't have done that. Fixed.